packages/@aws-cdk/aws-cognito-identitypool/lib/identitypool-role-attachment.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  CfnIdentityPoolRoleAttachment,
+} from '@aws-cdk/aws-cognito';
+import {
+  IRole,
+} from '@aws-cdk/aws-iam';
+import {
+  Resource,
+  IResource,
+} from '@aws-cdk/core';
+import {
+  Construct,
+} from 'constructs';
+import {
+  IIdentityPool,
+  IdentityPoolProviderUrl,
+} from './identitypool';
+
+/**
+ * Represents an Identity Pool Role Attachment
+ */
+export interface IIdentityPoolRoleAttachment extends IResource {
+  /**
+   * Id of the Attachments Underlying Identity Pool
+   */
+  readonly identityPoolId: string;
+}
+
+/**
+ * Props for an Identity Pool Role Attachment
+ */
+export interface IdentityPoolRoleAttachmentProps {
+
+  /**
+   * The name of the Identity Pool Role Attachment
+   * @default - automatically generated name by CloudFormation at deploy time
+   */
+  readonly identityPoolRoleAttachmentName?: string;
```

Comment:
It doesn't look like this is used. The CfnIdentityPoolRoleAttachment doesn't have a name property.

Reply:
Removed